src/main/java/com/palantir/docker/compose/DockerComposition.java

```diff
@@ -101,7 +102,11 @@ public static DockerCompositionBuilder of(String dockerComposeFile, DockerMachin
     }
 
     public static DockerCompositionBuilder of(DockerComposeFiles dockerComposeFiles, DockerMachine dockerMachine) {
-        return of(new DefaultDockerCompose(dockerComposeFiles, dockerMachine));
+        return of(new DefaultDockerCompose(dockerComposeFiles, dockerMachine, ProjectName.random()));
+    }
+
+    public static DockerCompositionBuilder of(DockerComposeFiles dockerComposeFiles, DockerMachine dockerMachine, String projectName) {
```

Comment:
My first instinct is that `projectName` should live on the builder but you need it here to pass through a docker compose. I really don't like this `of(...).builder().methods()` setup, I wish these `of` overloads had been methods on the builder itself.